samples/eShopLite/MyFrontend/Components/Cart.razor

```diff
@@ -18,27 +18,25 @@
 </div>
 
 @code {
+    CustomerBasket? customerBasket;
     bool basketIsAvailable;
-    int itemsInCart = 0;
 
     [Parameter]
     public EventCallback<bool> BasketAvailabilityChanged { get; set; }
 
     protected override async Task OnInitializedAsync()
     {
-        var (basket, isAvailable) = await BasketClient.GetBasketAsync("user");
+        (customerBasket, basketIsAvailable) = await BasketClient.GetBasketAsync("user");
 
-        if (basket is not null)
-        {
-            itemsInCart = basket.TotalItemCount;
-        }
-        basketIsAvailable = isAvailable;
         await BasketAvailabilityChanged.InvokeAsync(basketIsAvailable);
     }
 
     private async Task HandleCheckout()
     {
-        await BasketClient.CheckoutBasketAsync("user");
+        if (customerBasket is not null)
+        {
+            await BasketClient.CheckoutBasketAsync("user");
+        }
 
         // Preserve query string
         Navigation.NavigateTo($"/{new Uri(Navigation.Uri).Query}");
```

Comment:
Should the NavigateTo only be called if there is a customer basket as well?

Reply:
We're still in a POST request so we need to redirect (post-redirect-get).